features/steps/test_behave.py

```diff
@@ -1,108 +1,215 @@
+    return None
+
 
 def execute_dnf_command(cmd, reponame):
-  " Execute DNF command with default flags and the specified `reponame` enabled "
-  flags = DNF_FLAGS + ['--enablerepo={0}'.format(reponame)]
-  return subprocess.check_call(['dnf'] + flags + cmd, stdout=subprocess.PIPE)
+    """ Execute DNF command with default flags and the specified `reponame` enabled """
+    flags = DNF_FLAGS + ['--enablerepo={0}'.format(reponame)]
+    return subprocess.check_call(['dnf'] + flags + cmd, stdout=subprocess.PIPE)
+
+
+def execute_dnf_command_notinstall(cmd, reponame):
+    """ Execute DNF command with default flags and the specified `reponame` enabled """
+    flags = DNF_FLAGS + ['--enablerepo={0}'.format(reponame)]
+    return subprocess.call(['dnf'] + flags + cmd, stdout=subprocess.PIPE)
+
 
 def execute_rpm_command(pkg, action):
-  " Execute given action over specified pkg(s) "
-  if not isinstance(pkg, list):
-    pkg = [pkg]
-  if action == "remove":
-    action = RPM_ERASE_FLAGS
-  elif action == "install":
-    action = RPM_INSTALL_FLAGS
-    pkg = decorate_rpm_packages(pkg)
-  return subprocess.check_call(['rpm'] + action + pkg, stdout=subprocess.PIPE)
+    """ Execute given action over specified pkg(s) """
+    if not isinstance(pkg, list):
+        pkg = [pkg]
+    if action == "remove":
+        action = RPM_ERASE_FLAGS
+    elif action == "install":
+        action = RPM_INSTALL_FLAGS
+        pkg = decorate_rpm_packages(pkg)
+    return subprocess.check_call(['rpm'] + action + pkg, stdout=subprocess.PIPE)
+
 
 def piecewise_compare(a, b):
-  " Check if the two sequences are identical regardless of ordering "
-  return sorted(a) == sorted(b)
+    """ Check if the two sequences are identical regardless of ordering """
+    return sorted(a) == sorted(b)
+
 
 def split(pkg):
-  return [p.strip() for p in pkg.split(',')]
+    return [p.strip() for p in pkg.split(',')]
+
 
 @given('I use the repository "{repo}"')
 def given_repo_condition(context, repo):
-  " :type context: behave.runner.Context "
-  assert repo
-  assert os.path.exists('/build/' + repo)
-  a = [os.remove(p) for p in os.listdir('/repo')]
-  subprocess.check_call(['cp -rs /build/' + repo + '/* /repo/'], shell=True)
+    """ :type context: behave.runner.Context """
+    assert repo
+    context.repo = repo
+    assert context.repo
+    assert os.path.exists('/var/www/html/repo/' + repo)
+    subprocess.check_call(['echo -ne "[' + repo + ']\nname=' + repo + '\nbaseurl=http://127.0.0.1/repo/' + repo + '\nenabled=1\ngpgcheck=0" > /etc/yum.repos.d/' + repo + '.repo'], shell=True)
+
 
 @when('I "{action}" a package "{pkg}" with "{manager}"')
 def when_action_package(context, action, pkg, manager):
-  assert action in ["install", "remove"]
-  assert manager in ["rpm", "dnf", "pkcon"]
-  assert pkg
-  context.pre_packages = get_package_list()
-  assert context.pre_packages
+    assert action in ["install", "remove", "upgrade", "downgrade", "notinstall", "autoremove", "upgrade-to"]
```

Comment:
we should refactor this. It's too complicated and limiting (for one package). It should just take dnf command line. (this could be fixed in later commits)